bigquery_etl/view/__init__.py

```diff
@@ -176,27 +177,45 @@ def udf_references(self):
 
     @cached_property
     def view_schema(self):
-        """Derive view schema from a schema file or a dry run result."""
-        schema_file = Path(self.path).parent / "schema.yaml"
+        """Derive view schema from a dry run result and/or a schema file."""
+        schema = None
+
+        # check schema based on dry run results
+        try:
+            client = bigquery.Client()
+            query_job = client.query(
+                # We have to remove `CREATE OR REPLACE VIEW ... AS` from the query to avoid
+                # view creation permission denied errors, and we have to apply a `WHERE FALSE`
+                # filter to avoid partition column filter missing errors.
+                query=dedent(
+                    f"""
+                    WITH view_query AS (
+                        {CREATE_VIEW_PATTERN.sub("", self.content)}
+                    )
+                    SELECT *
+                    FROM view_query
+                    WHERE FALSE
+                    """
+                ),
```

Comment:
I was pleasantly surprised that dry-running the full `CREATE VIEW ...` query returns the view's schema, nicely side-stepping the issue of potentially having to add query filters on any partition columns from the underlying tables.  However, it turns out that only works if you actually have permission to create the view in the target dataset, which we DEs generally do but CI does not, so the previous dry-run-to-get-view-schema logic would have always failed in a CI context.